### ISSUE TITLE
auth/acl evaluation order with multiples auth plugins ?

### ISSUE THREAD

**Issue by thpham** [emqtt/emqttd#889]
#### Environment

- OS: macOS 10.12.3 (16D32)
- Erlang/OTP:
- EMQ: 2.0.7

#### Description

I'm using the acl.conf with the plugins emq_auth_username.conf and emq_auth_http.conf.
It seems not possible to combine the acl with username plugin and having the auth_http as a fallback.

Is it related to loaded_plugins order or something ? Please keep me posted asap.

Thank you.

**Comment by thpham** [comment]
I still have the same problem when I use emq_auth_redis.conf and emq_auth_http.conf in the same time. The latest plugin loaded is used for the authentication.

Would be great to have both used to check auth/acl.

**Comment by emqplus** [comment]
@thpham The issue has been fixed in 2.2-beta.1 release. We improved the Auth/ACL chain design in version 2.2. The Auth request will be forwarded to next auth module if it is ignored by the current auth module:

```

               --------------           -------------           --------------
    Client --> | Redis Auth | -ignore-> | HTTP Auth | -ignore-> | MySQL Auth |
               --------------           -------------           --------------
                     |                       |                       |
                    \|/                     \|/                     \|/
               allow | deny            allow | deny            allow | deny
```
